### ISSUE TITLE
About upsample, the kernel_size is 1024?

### ISSUE THREAD

**Issue by hdmjdp** [NVIDIA/waveglow#23]
self.upsample = torch.nn.ConvTranspose1d(n_mel_channels,
                                                 n_mel_channels,
                                                 1024, stride=256)

I think the kernel_size is too large

**Comment by dhgrs** [comment]
Transposed convolution needs larger kernel size than convolution for the same size reseptive field.
Check this repository!
https://github.com/vdumoulin/conv_arithmetic

**Comment by hdmjdp** [comment]
@can you explain in detail? thanks

**Comment by dhgrs** [comment]
Transposed convolution is combination of 2 operations. First is upsampling and second is convolution.
The upsampling is done with zero-padding between pixels. So that convolution's kernel is multiplied with very sparse vector. And we need to increase the size of kernel to get large receptive field.

**Comment by rafaelvalle** [comment]
Note that the ConvTranspose1d parameters are directly related to the STFT parameters, whereas kernel_size is equal to filter_length and stride is equal to hop_size. Matching these parameters we are able to upsample the dimensionality of the mel-spectrogram to the dimensionality of the audio that was used to compute the mel-spectrogram.

**Comment by wizardk** [comment]
> Note that the ConvTranspose1d parameters are directly related to the STFT parameters: kernel_size is equal to filter_length and stride is equal to hop_size.
> 
> Matching these parameters we are able to upsample the dimensionality of the mel-spectrogram to the dimensionality of the audio that was used to compute the mel-spectrogram.

kernel_size is equal to win_length, not filter_length
